### PR TITLE
ruff patch for one-time linter/formatter catchup

### PR DESCRIPTION
**Pull request by anthonywu**
Following the `ruff format` and `ruff check` contributor proposal in #50, this would be the one time big patch to cleanse the repo state, from which point a pre-commit hook or a server-side CI check can be introduced to enforce future deviations.

If accepted, we should allow a few more ongoing PRs to merge to `main` first, then come back to this branch to `rebase` and do another catch up format/check.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -15,16 +15,16 @@ classifiers = [
     "Operating System :: MacOS",
 ]
 dependencies = [
+    "huggingface-hub>=0.24.5",
```
Comment (anthonywu) [inline pyproject.toml:18]: these changes are from #50 - the diff would disappear in this PR if/when #50 merges

File: src/mflux/config/config.py [hunk 2]
```diff
@@ -9,11 +9,11 @@ class Config:
...
-            self,
-            num_inference_steps: int = 4,
-            width: int = 1024,
-            height: int = 1024,
-            guidance: float = 4.0,
+        self,
+        num_inference_steps: int = 4,
+        width: int = 1024,
+        height: int = 1024,
+        guidance: float = 4.0,
```
Comment (anthonywu) [inline src/mflux/config/config.py:16]: lots of these changes coming from `ruff format`

File: src/mflux/post_processing/image.py [hunk 3]
```diff
@@ -77,33 +76,18 @@ def _embed_metadata(self, path: str) -> None:
...
-            exif_dict = {
-                '0th': {},
-                'Exif': {
-                    USER_COMMENT_TAG_ID: user_comment_bytes
-                },
-                'GPS': {},
-                '1st': {},
-                'thumbnail': None
-            }
-
```
Comment (anthonywu) [inline src/mflux/post_processing/image.py:95]: caught by `ruff check` - `exif_dict` was never used

File: src/mflux/weights/lora_converter.py [hunk 4]
```diff
@@ -25,7 +25,6 @@ def load_weights(lora_path: str) -> dict:
     def _load_pytorch_weights(lora_path: str) -> dict:
         state_dict = {}
         with safe_open(lora_path, framework="pt") as f:
-            metadata = f.metadata()
```
Comment (anthonywu) [inline src/mflux/weights/lora_converter.py:28]: also from `ruff check` - `metadata` was not used

File: src/mflux/post_processing/image_util.py [hunk 5]
```diff
@@ -1,5 +1,4 @@
 import PIL
-from PIL import Image
```
Comment (anthonywu) [inline src/mflux/post_processing/image_util.py:2]: caught by `ruff check`, this `PIL.Image` object was immediately shadowed a few lines later by `from mflux.post_processing.image import Image`

**Review by anthonywu — COMMENTED** [review]
a disruptive, and hopefully a no-op PR, trusting the authors of `ruff` to have done a good job with the automatic edits

obviously needs testing with basic use cases prior to merging

**Comment by anthonywu** [comment]
this PR is easy to re-generate after a `rebase`, waiting for `main` dust to settle first then will re-approach

**Comment by anthonywu** [comment]
Closing this preview PR for now, this big diff should be re-done following agreement in #58

@filipstrand maybe you ought to be the first user of the ruff configs, own this big change, and keep the `git` line ownership to you. It would be less confusing on line-level contributor stats =)

**Comment by filipstrand** [comment]
@anthonywu Yes, I can be the one who ultimately applies the formatting updates :) I may make some small tweaks to the ruff setup also, but from my initial testing the ones in #58 looks very reasonable.

**Pull request closed without merging**